### ISSUE TITLE
Invalid address to set: []string{"integrations_server", "0", "symbols_https_endpoint"}  or Invalid address to set: []string{"integrations_server", "0", "profiling_https_endpoint"}

### ISSUE THREAD

**Issue by davidg-datascene** [elastic/terraform-provider-ec#781]
- [ x] I am running the latest version
- [ x] I checked the documentation and found no answer
- [ x] I checked to make sure that this issue has not already been filed
- [ x] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior
The terraform plan runs through without error

## Terraform definition
```
# Create an Elastic Cloud deployment
resource "ec_deployment" "elasticcloud" {
  # Optional name.
  name = "my_example_deployment"

  region                 = "us-east-1"
  version                = data.ec_stack.latest.version
  deployment_template_id = "aws-io-optimized-v2"

  elasticsearch = {
    hot = {
      autoscaling = {}
    }
  }

  kibana = {}

  enterprise_search = {}

  integrations_server = {}
}
```
## Steps to Reproduce
ElasticCloud have introduced some new endpoints (Profiling and Symbols) as per screen shot.

EC version: 8.11.3

![Screenshot 2024-02-13 at 12 50 53](https://github.com/elastic/terraform-provider-ec/assets/57825578/400b78b5-a6da-4f94-b6fe-6de15e9e4781)

which is causing the plan to fail on subsequent runs (symbols_https_endpoint or profiling_https_endpoint):

```console
Error: Invalid address to set: []string{"integrations_server", "0", "symbols_https_endpoint"}

  with module.main.module.elastic_cluster[0].ec_deployment.elasticcloud,
  on .terraform/modules/main.elastic_cluster/elastic.tf line 7, in resource "ec_deployment" "elasticcloud":
   7: resource "ec_deployment" "elasticcloud" {
   
 ```

**Comment by davidg-datascene** [comment]
Might need to be added to the [Nested Schema for integrations_server.endpoints](https://registry.terraform.io/providers/elastic/ec/latest/docs/resources/deployment#nested-schema-for-integrations_serverendpoints)

Thanks.

**Comment by davidg-datascene** [comment]
Apologies to pester but this is impacting use hard i.e. we can't apply elastic changes through our terraform code beacuse of the issue. Thanks.

**Comment by tobio** [comment]
@davidg-datascene I'm struggling to reproduce this with just the resource definition included here. Are you able to produce a minimal set of steps to reproduce this error? Specifically covering:

1. Terraform versions `terraform version` from within your TF module dir would be great, it would include both the CLI and provider versions. 
2. The steps you're taking to enable the symbols/profiling URLS, IIUC these would only be available after enabling universal profiling. 
3. Anything else required to go from a clean TF state to a failing state.

**Comment by davidg-datascene** [comment]
We have introduced APM (integrations_server) into Kibana via the terraform resource "ec_deployment" as shown by the snippet below.  We have a nonprod and prod cluster using APM (integrations_server) built using the same terraform module.  On the nonprod cluster the issue doesn't exist, probably beacuse the Symbols and Profiling endpoints haven't been activated. We haven't introduced Universal Profiling on either cluster but suspect someone may have clicked on the Universal Profiling link in the Observability part of the PROD Kibana to activate the endpoints. 

Elastic cloud version: 8.11.3

Terrafrom provider version (we currently can't upgrade to the latest EC provider but I suspect the issue is with the latest provider as well)

```console
terraform {
  required_providers {
    ec = {
      source  = "elastic/ec"
      version = "< 0.6"
    }
  }
  required_version = ">= 0.14"
}
```
## Include APM (integrations_server) in Cluster deployment
```console
resource "ec_deployment" "elasticcloud" {
        name                   = "test-with-apm"
       
### Include integrations_server (APM)
      integrations_server {
         topology {
                 size = "4g"
                zone_count   = 3
            }
        }
    }
}
```
Let me know if you need more info. Thanks

**Comment by 3kt** [comment]
@tobio the below config can be copy-pasted to reproduce the issue:

Run the below `main.tf`:

```
terraform {
  required_providers {
    ec = {
      source  = "elastic/ec"
      version = "< 0.6"
    }
  }
  required_version = ">= 0.14"
}

provider "ec" {
    apikey = "YOUR_API_KEY"
}

resource "ec_deployment" "elasticcloud" {
  name = "test-with-apm"
  version = "8.11.3"
  deployment_template_id = "azure-storage-optimized"
  region                 = "azure-northeurope"

  elasticsearch {}

  kibana {}

  integrations_server {
     topology {
            size = "4g"
            zone_count   = 1
        }
    }
}
```

Once deployment is finished, you can verify that a `terraform refresh` completes without issue:

```
❯ terraform refresh
ec_deployment.elasticcloud: Refreshing state... [id=ID]
```

If you then navigate to https://YOUR_DEP.kb.northeurope.azure.elastic-cloud.com:9243/app/profiling and click on "Set up universal profiling", any subsequent refresh will fail because of the added endpoint, per the initial description:

<img width="561" alt="image" src="https://github.com/elastic/terraform-provider-ec/assets/12101744/6e9ab472-02b7-44b7-aede-c6cb5319b764">

<img width="434" alt="image" src="https://github.com/elastic/terraform-provider-ec/assets/12101744/689b8e33-c103-4c92-8fef-b9fa48b969e8">


```
❯ terraform refresh
ec_deployment.elasticcloud: Refreshing state... [id=ID]
╷
│ Error: Invalid address to set: []string{"integrations_server", "0", "symbols_https_endpoint"}
│
│   with ec_deployment.elasticcloud,
│   on main.tf line 15, in resource "ec_deployment" "elasticcloud":
│   15: resource "ec_deployment" "elasticcloud" {
│
╵
```

**Comment by tobio** [comment]
@3kt @davidg-datascene thanks for the repro steps. This issue has been fixed in 0.6+. 

Is there another issue blocking an upgrade?

**Comment by davidg-datascene** [comment]
> @3kt @davidg-datascene thanks for the repro steps. I can repro this with 0.5, but this issue has been fixed in 0.6+.
> 
> Is there another issue blocking an upgrade?

@tobio - Hi, thanks for reproducing it. We are looking at the options to upgrade to v6 as that is a breaking change and then from there to the latest.
